web/modules/custom/weather/src/Services/WeatherApi.php

```diff
@@ -39,15 +47,54 @@ public function validateCityApi(string $cityName): bool {
     $config = $this->configFactory->get('weather.settings');
     $apiKey = $config->get('api_admin_key');
     try {
-      $client = $this->httpClient->get('http_client');
-      $res = $client->get('https://api.openweathermap.org/data/2.5/weather?q=' . $cityName . '&units=metric&appid=' . $apiKey);
-      $body = (string) $res->getBody();
-      json_decode($body, TRUE);
+      $res = $this->httpClient->request('GET', 'https://api.openweathermap.org/data/2.5/weather?q=' . $cityName . '&units=metric&appid=' . $apiKey);
     }
-    catch (\Exception $error) {
+    catch (GuzzleException $e) {
       return FALSE;
     }
     return TRUE;
   }
 
+  /**
+   * Get the value from OpenWeather.
+   */
+  public function getWeatherValue(string $city, $key): array {
```

Comment:
$key is unnecessary if it 100% time will have same value that doesn't depend on calling code.